src/nvim/tui/tui.c

```diff
@@ -1820,6 +1838,21 @@ static void augment_terminfo(TUIData *data, const char *term,
       ut, "ext.enable_mouse", "\x1b[?1002h\x1b[?1006h");
   data->unibi_ext.disable_mouse = (int)unibi_add_ext_str(
       ut, "ext.disable_mouse", "\x1b[?1002l\x1b[?1006l");
+
+  if ((int)unibi_add_ext_bool(ut, "Su", 0)) {
+      data->unibi_ext.enter_undercurl_mode = (int)unibi_add_ext_str(
+          ut, "ext.enter_undercurl_mode", "\x1b[4:3m");
+      data->unibi_ext.exit_undercurl_mode = (int)unibi_add_ext_str(
+          ut, "ext.exit_underline_mode", "\x1b[4:0m");
+      // TODO: copy from setab and replace 48 with 58?
```

Comment:
Yes, mainly to avoid duplication, but I have not really checked if it matches the other conditionals.
kitty's docs just said that it works like `38`/`48`, but terminals appear to handle this differently.
It is likely ok though.